apps/src/code-studio/pd/workshop_survey/VariableFormGroup.jsx

```diff
@@ -0,0 +1,148 @@
+import React from 'react';
+
+import RadioFormGroup from './RadioFormGroup';
+import CheckboxFormGroup from './CheckboxFormGroup';
+
+const SINGLE_SELECT = 'single_select';
+const MULTI_SELECT = 'multi_select';
+const FREE_RESPONSE = 'free_response';
+
+const questionPropType = React.PropTypes.shape({
+  label: React.PropTypes.string.isRequired,
+  name: React.PropTypes.string.isRequired,
+  required: React.PropTypes.bool,
+  type: React.PropTypes.oneOf([SINGLE_SELECT, MULTI_SELECT, FREE_RESPONSE]).isRequired,
+  values: React.PropTypes.arrayOf(React.PropTypes.string),
+});
+
+const ColumnVariableQuestion = React.createClass({
+  propTypes: {
+    selectedValues: React.PropTypes.arrayOf(React.PropTypes.string).isRequired,
+    question: questionPropType
+  },
+
+  buildColumn(selectedValue) {
+    const key = `${this.props.question.name}[${selectedValue}]`;
+
+    let formGroup;
+    if (this.props.question.type === SINGLE_SELECT) {
+      formGroup = <RadioFormGroup name={key} values={this.props.question.values} />;
+    } else if (this.props.question.type === MULTI_SELECT) {
+      formGroup = <CheckboxFormGroup name={key} values={this.props.question.values} />;
+    }
```

Comment:
Note that right now, columns only support single- or multi-select options, and rows only support free response. This is slightly more than we need for the current implementation, but both could reasonably be expanded to support more types for future needs.